ir/ir.rs

```diff
@@ -513,6 +513,15 @@ pub enum ItemKind {
 
     /// Miscellaneous item. Perhaps metadata. Perhaps something else.
     Misc(Misc),
+
+    /// Program scope. Modules, namespaces, imported declarations.
+    Scope(Scope),
+
+    /// Subroutine item.
+    Subroutine(Subroutine),
+
+    /// Type entry.
+    Type(Type),
```

Comment:
Types (ignoring RTTI) don't really exist at runtime or in the binary. Wasm has its types section, but each type entry is pretty tiny and I'd consider it something that could be `Misc`. What is the motivation for this new item kind?